docs/src/main/asciidoc/optaplanner.adoc

```diff
@@ -0,0 +1,1034 @@
+        timeslotList.add(new Timeslot(DayOfWeek.MONDAY, LocalTime.of(9, 30), LocalTime.of(10, 30)));
+        timeslotList.add(new Timeslot(DayOfWeek.MONDAY, LocalTime.of(10, 30), LocalTime.of(11, 30)));
+        timeslotList.add(new Timeslot(DayOfWeek.MONDAY, LocalTime.of(13, 30), LocalTime.of(14, 30)));
+        timeslotList.add(new Timeslot(DayOfWeek.MONDAY, LocalTime.of(14, 30), LocalTime.of(15, 30)));
+
+        List<Room> roomList = new ArrayList<>();
+        roomList.add(new Room("Room A"));
+        roomList.add(new Room("Room B"));
+        roomList.add(new Room("Room C"));
+
+        List<Lesson> lessonList = new ArrayList<>();
+        lessonList.add(new Lesson(101L, "Math", "B. May", "9th grade"));
+        lessonList.add(new Lesson(102L, "Physics", "M. Curie", "9th grade"));
+        lessonList.add(new Lesson(103L, "Geography", "M. Polo", "9th grade"));
+        lessonList.add(new Lesson(104L, "English", "I. Jones", "9th grade"));
+        lessonList.add(new Lesson(105L, "Spanish", "P. Cruz", "9th grade"));
+
+        lessonList.add(new Lesson(201L, "Math", "B. May", "10th grade"));
+        lessonList.add(new Lesson(202L, "Chemistry", "M. Curie", "10th grade"));
+        lessonList.add(new Lesson(203L, "History", "I. Jones", "10th grade"));
+        lessonList.add(new Lesson(204L, "English", "P. Cruz", "10th grade"));
+        lessonList.add(new Lesson(205L, "French", "M. Curie", "10th grade"));
+        return new TimeTable(timeslotList, roomList, lessonList);
+    }
+
+}
+----
+
+This test verifies that after solving, all lessons are assigned to a time slot and a room.
+It also verifies that it found a feasible solution (no hard constraints broken).
+
+Add a test properties to the `src/main/resources/application.properties` file:
+
+----
+# The solver runs only for 5 seconds to avoid a HTTP timeout in this simple implementation.
+# It's recommended to run for at least 5 minutes ("5m") otherwise.
+quarkus.optaplanner.solver.termination.spent-limit=5s
+
+# Effectively disable this termination in favor of the best-score-limit
+%test.quarkus.optaplanner.solver.termination.spent-limit=1h
+%test.quarkus.optaplanner.solver.termination.best-score-limit=0hard/*soft
+----
+
+Normally, the solver finds a feasible solution in less than 200 milliseconds.
+Notice how the `application.properties` overwrites the solver termination during tests
+to terminate as soon as a feasible solution (`0hard/*soft`) is found.
+This avoids hard coding a solver time, because the unit test might run on arbitrary hardware.
+This approach ensures that the test runs long enough to find a feasible solution, even on slow machines.
+But it does not run a millisecond longer than it strictly must, even on fast machines.
+
+=== Logging
+
+When adding constraints in your `ConstraintProvider`,
+keep an eye on the _score calculation speed_ in the `info` log,
+after solving for the same amount of time, to assess the performance impact:
+
+----
```

Comment:
Source block.

Reply:
@gsmet This is console output (logging). What is the best way to format that?
A) Nothing, it's not source code
B) `[source,shell]` What happens if that contains a log line with shell keywords?
C) `[source]`

Reply:
You can just use `[source]`. We don't have proper formatting on the website if we don't have that.